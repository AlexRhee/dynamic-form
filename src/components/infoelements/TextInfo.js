import React from 'react';

//Element for the text info (post submission)
const TextInput = (props) => {
    return (
        <div data-testid="textinfo-1" className="input-wrapper">
            <label>{props.field.name}:</label>
            <div className="inner-wrapper">
                <input disabled type="text" name={props.field.id} className="form-control" aria-describedby="emailHelp" placeholder={props.field.placeholder} value={props.field.value}/>
            </div>
        </div>
    )
}

export default TextInput;