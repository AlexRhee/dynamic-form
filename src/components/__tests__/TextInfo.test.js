import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import TextInfo from '../infoelements/TextInfo';


afterEach(() => {
    cleanup();
})

test('should render TextInfo component and be disabled', () => {

    const field =         {
        "id": "firstName",
        "name": "First Name",
        "placeholder": "First name",
        "required": true,
        "type": "text",
        "ind": 0,
        "value": "Alex"
    }

    render(<TextInfo field={field} />);
    const textinfoElement = screen.getByTestId('textinfo-1');
    expect(textinfoElement).toBeInTheDocument();
    expect(textinfoElement).toHaveTextContent('First Name');
    expect(textinfoElement).toBeDisabled;
})

test('matches snapshot', () => {

    const field =         {
        "id": "firstName",
        "name": "First Name",
        "placeholder": "First name",
        "required": true,
        "type": "text",
        "ind": 0,
        "value": "Alex"
    }

    const tree = renderer.create(<TextInfo field={field} />).toJSON();
    expect(tree).toMatchSnapshot();
})