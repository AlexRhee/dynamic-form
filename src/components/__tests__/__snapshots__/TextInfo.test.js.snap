// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="input-wrapper"
  data-testid="textinfo-1"
>
  <label>
    First Name
    :
  </label>
  <div
    className="inner-wrapper"
  >
    <input
      aria-describedby="emailHelp"
      className="form-control"
      disabled={true}
      name="firstName"
      placeholder="First name"
      type="text"
      value="Alex"
    />
  </div>
</div>
`;
